compiler_opt/es/blackbox_optimizers.py

```diff
@@ -31,15 +31,21 @@
 import abc
 import math
 import numpy as np
+import numpy.typing as npt
 from sklearn import linear_model
 from absl import flags
 import scipy.optimize as sp_opt
+from typing import List, Dict, Callable, Tuple, Any, Optional, Union, Literal, Mapping, Sequence
 
 from compiler_opt.es import gradient_ascent_optimization_algorithms
 
 
-def filter_top_directions(perturbations, function_values, est_type,
-                          num_top_directions):
+def filter_top_directions(
+    perturbations: npt.NDArray[np.float32],
+    function_values: npt.NDArray[np.float32],
+    est_type: Union[Literal['forward_fd'],
```

Comment:
We should change `est_type` to an enum. Could you add an issue about that and a TODO here so we don't forget?

Reply:
Is it ok if I change it to an enum right now instead?

Reply:
It has been changed to an enum now